index.bs

```diff
@@ -2953,6 +2953,282 @@ value and terminate the operation.
 
 </div>
 
+### Signal Credential Changes to the Authenticator - PublicKeyCredential's [=signal methods=] ### {#sctn-signal-methods}
+
+<xmp class="idl">
+    partial interface PublicKeyCredential {
+        static Promise<undefined> signalUnknownCredentialId(UnknownCredentialIdOptions options);
+        static Promise<undefined> signalAllAcceptedCredentialIds(AllAcceptedCredentialIdsOptions options);
+        static Promise<undefined> signalCurrentUserDetails(CurrentUserDetailsOptions options);
+    };
+
+    dictionary UnknownCredentialIdOptions {
+        required DOMString                     rpId;
+        required Base64URLString               credentialId;
```

Comment:
Ah good catch. If the get request was in a non-discoverable flow, this can indeed happen depending on how the RP is implemented. (It would happen and be a problem on Google.com with our current implementation)

If a credential is rejected after a discoverable flow, should the RP just pass the userId would just come from the assertion? Or do we need this report both with and without userId?

(Semi-related: In initial drafts we had a whole other report that said "this userId is invalid or gone" - implying that all credentials for that userId could be deleted. But that's probably less useful.)

Reply:
> Scenario: A user has multiple accounts on a service, perhaps protected by two different physical authenticators. When authenticating to account `a`, they try to use the authenticator for account `b`

> If the get request was in a non-discoverable flow, this can indeed happen depending on how the RP is implemented. 

If the relying party knows which account is expecting an assertion from, the request should have an `allowedCredentials` list populated with that user's list of credential ids. The user trying to use the wrong authenticator will not result in an assertion being passed to the relying party, so the relying party doesn't have any reason to call `signalUnknownCredentialId`.

This scenario would only be possible if the relying party chose to send an empty allow-list. But why would any relying party risk having the user be shown passkeys for a different account? Non-discoverable credentials are not even required to store the `userId`.

Reply:
@nsatragno is right here, I forgot that in a non-discoverable flow an unrecognized credential wouldn't happen without some other kind of error on the RP's behalf, because otherwise that credentialId shouldn't have been on the allow list.

So an unrecognized credential can only happen if the RP doesn't know what account they're expecting. So no need for scoping the credentialId in the report.

Reply:
Agreed. While the wording does technically state that allowCredentials SHOULD rather than MUST be populated, I find it hard to think of a reasonable situation that would both be empty and result in an assertion being generated (I _think_ it can happen on some stateless authenticators, but by their nature they'd ignore this signal anyway).

Thanks for the added perspective. Consider this resolved (I don't seem to have the option to do so myself, despite starting the thread)